### DISCUSSION TITLE
Testing configuration and organization

### DISCUSSION THREAD

**Discussion by pratishta** [NYCPlanning/ae-zoning-api#169] [General]
# Description
Discussion spawned from comment [here](https://github.com/NYCPlanning/ae-zoning-api/pull/140#discussion_r1455919249). 

We should figure out how we want to:

1. Organize our test files 
2. Run our test suites 

### Organizing the test files
Currently we have unit tests as individual `spec` files within the domain directory (e.g. `src/tax-lot/tax-lot.service.spec.ts`) and e2e tests within the top level `test` directory with its own domain subfolders and mock files (e.g. `test/tax-lot`). This is the default way in Nest to test but we should have a more cohesive way to organize the files so that it is easier to share data and mocks and also more intuitive to run tests during development.

### Running our tests
Our current two testing commands `npm run test` and `npm run test:e2e` run the unit and e2e tests respectively. They don't show the verbose outputs in either but that may be helpful to have in certain cases. When we run tests for a specific case e.g. `npm run test:e2e test/tax-lot` verbose output is on by default. 

It'll be helpful to determine how we expect to run our tests so that we can decide how we want to configure Jest for development vs CI, etc

**Comment by TangoYankee** [comment]
We're also running into these questions for our front-end test suites. [adding playwright to the remix template](https://github.com/NYCPlanning/ae-remix-template/pull/6#discussion_r1521968129)
